### PR TITLE
Add search button to LocationAutocomplete

### LINKED ISSUE #1403: Search issues
Subject: Search issues
Description:
Tried to search  Search filters using "near London" as example

Results:
Seems to expect drop-down location  box which is not there. Entered "London" manually and results returned included people in other countries. How do I search for a host in London ? Is there any "Help" functionality ?

Backend version: develop-762ac12b
Frontend version: develop-762ac12b
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.77 Safari/537.36
Page: https://app.couchers.org/search#loc=1/0/0
User (spoofable): paulpd (429)

### PR DESCRIPTION
**Pull request by lucaslcode**
closes #1403 

**Frontend checklist**
- [x] Formatted my code with `yarn format && yarn lint --fix`
- [x] There are no warnings from `yarn lint`
- [x] There are no console warnings when running the app
- [ ] Added any new components to storybook
- [x] Added tests where relevant
- [x] All tests pass
- [x] Clicked around my changes running locally and it works
- [x] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

File: app/frontend/src/components/Autocomplete.tsx [hunk 1]
```diff
@@ -56,6 +58,19 @@ export default function Autocomplete<
...
           label={label}
           helperText={error || helperText}
+          InputProps={
+            endAdornment
+              ? {
+                  ...params.InputProps,
+                  endAdornment: (
+                    <>
+                      {params.InputProps.endAdornment}
+                      {endAdornment}
```
Comment (darrenvong) [inline app/frontend/src/components/Autocomplete.tsx:68]: Is one of these the cross for clearing the input field?

File: app/frontend/src/features/search/LocationAutocomplete/LocationAutocomplete.tsx [hunk 2]
```diff
@@ -90,6 +97,15 @@ export default function LocationAutocomplete({
           searchSubmit(controller.field.value, "create-option");
         }
       }}
+      endAdornment={
```
Comment (darrenvong) [inline app/frontend/src/features/search/LocationAutocomplete/LocationAutocomplete.tsx:100]: It'd be good to add a test to check clicking the button submits just like pressing enter

**Review by darrenvong — CHANGES_REQUESTED** [review]
Mostly looking good apart from the very minor comments

File: app/frontend/src/components/Autocomplete.tsx (hunk 1 above)
Comment (lucaslcode) [reply app/frontend/src/components/Autocomplete.tsx:68]: yes, or whatever mui autocomplete wants to put there

**Review by darrenvong — APPROVED** [review]

**Pull request merged**